src/cursor/common.rs

```diff
@@ -98,63 +105,131 @@ where
             client: self.client,
             provider: self.provider,
             buffer: self.buffer,
+            post_batch_resume_token: self.post_batch_resume_token,
             info: self.info,
             pinned_connection: self.pinned_connection,
             _phantom: Default::default(),
         }
     }
 }
 
-impl<P, T> Stream for GenericCursor<P, T>
+pub(crate) trait CursorStream {
```

Comment:
This trait allows the implementation of `poll_next` to be shared via the `stream_poll_next` fn; open to suggestions for a more descriptive name.